packages/vuetify/src/components/VDataTable/VDataTable.ts

```diff
@@ -449,6 +466,7 @@ export default VDataIterator.extend({
         key: getObjectValueByPath(item, this.itemKey),
         class: {
           ...classes,
+          ...this.createItemClass(this.itemClass(item)),
```

Comment:
maybe we could use mergeData here (and skip the `createItemClass` part, not sure how much less performant it would be

Reply:
If I understand how it works, 'class' item should be a key{str}: value{bool}, and mergeData returns an array, so I don't know how to do just with it?

Reply:
with `mergeData` you should pass the whole `VNodeData` object, but you can wait for https://github.com/vuetifyjs/vuetify/pull/11271

Reply:
Ok I wait for that merge, so to eventually apply this change, so. All other reviews are fixed now

Reply:
White `mergeClasses` you should be able to remove `createItemClass` and just do 
```js
class: mergeClasses(classes, this.itemClass(item))
```

Reply:
@florealcab merge, see example usage in https://github.com/vuetifyjs/vuetify/commit/9257de2b4c7960d1fd8846355aa54b9f5048bb12

Reply:
It works well, I remove the createItemClass method and use the new mergeClasses method instead! So fixed